research/auth/current-implementation.md

```
@@ -0,0 +1,144 @@
+## Overview
+
+The auth middleware now supports two spec-compliant OAuth flows:
+
+1. **`oauth`** -- OAuth 2.0 Authorization Code + PKCE (RFC 7636 + RFC 8252)
+2. **`device-code`** -- OAuth 2.0 Device Authorization Grant (RFC 8628)
+
+The previous non-standard flow (direct token delivery via POST to a local server) has been removed entirely.
+
+## OAuth Authorization Code + PKCE
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)